api/upsert_feature_set.rb

```diff
@@ -0,0 +1,234 @@
+  end
+
+  Parallel.each(file_paths, in_threads: 6) do |file_path|
+    filename = File.basename(file_path)
+    file_upload = file_uploads_map[filename]
+
+    uri = URI(file_upload['upload_url'])
+    file_contents = File.read(file_path)
+    Net::HTTP.start(uri.host) do |http|
+      puts "Upload #{file_path} to S3"
+      response = http.send_request('PUT',
+                                   uri,
+                                   file_contents,
+                                   file_upload['headers'])
+      puts "Done uploading #{file_path}, response.code = #{response.code}"
+    end
+  end
+end
+
+#
+# This method demonstrates how to use the IDs of the files that
+# were previously uploaded to Sentera's cloud storage with the
+# upsert_feature_set GraphQL mutation, to attach the files to
+# a feature set.
+#
+# @param [string] survey_sentera_id Sentera ID of the survey
+#                                   within FieldAgent that is
+#                                   the parent of the feature set
+# @param [string] feature_set_sentera_id ID of the feature set
+#                                        to upsert
+# @param [Array[Object]] file_uploads FileUpload GraphQL
+#                        objects created by the
+#                        create_file_uploads mutation
+#
+# @return [Hash] Hash containing results of the GraphQL request
+#
+def upsert_feature_set(geometry:, feature_set_sentera_id:, survey_sentera_id:, file_uploads:)
+  puts 'Upsert feature set'
+
+  gql = <<~GQL
+    mutation UpsertFeatureSet(
+      $feature_set: FeatureSetImport!
+      $owner: FeatureSetOwnerInput!
+    ) {
+      upsert_feature_set(
+        feature_set: $feature_set
+        owner: $owner
+      ) {
+        succeeded {
+          ... on FeatureSet {
+            sentera_id
+            name
+            type
+            status
+            released
+          }
+        }
+        failed {
+          attributes {
+            key
+            details
+            attribute
+          }
+        }
+      }
+    }
+  GQL
+
+  variables = {
+    owner: {
+      owner_type: 'SURVEY',
+      sentera_id: survey_sentera_id
+    },
+    feature_set: {
+      sentera_id: feature_set_sentera_id,
+      geometry: geometry,
+      annotation_file_keys: file_uploads.map { |file_upload| file_upload['id'] }
+    }
+  }
+
+  response = make_graphql_request(gql, variables)
+  json = JSON.parse(response.body)
+  json.dig('data', 'upsert_feature_set')
+end
+
+# MAIN
+
+# **************************************************
+# Set these variables based on the files you want
+# to upload and the survey within FieldAgent to
+# which you want to attach a feature set
+files_path = ENV.fetch('FILES_PATH', '.') # Your fully qualified path to a folder containing the files to upload
+file_ext = ENV.fetch('FILE_EXT', '*.*') # Your file extension
+geometry_path = ENV.fetch('GEOMETRY_PATH', 'test.geojson') # Your fully qualified path to file containing the feature set geometry
```

Comment:
@denisahearn 
I had to move test.geojson to the `api` folder for this variable to be built.  Can you update with the proper path?